### PR TITLE
First commit

### PR DESCRIPTION
**Pull request by deepakshingavi**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  "org.scalaj" % "scalaj-http_2.12" % "2.4.2",
+  "io.spray" %% "spray-json" % "1.3.5" % "provided",
+  "org.apache.spark" % "spark-core_2.12" % "2.4.4",
+  "org.apache.spark" % "spark-sql_2.12" % "2.4.4",
+  "org.scalatest" % "scalatest_2.12" % "3.3.0-SNAP2" % "provided",
+  "org.scalamock" % "scalamock_2.12" % "4.4.0"
+)
+
+assemblyMergeStrategy in assembly := {
+  case "mozilla/public-suffix-list.txt" => MergeStrategy.discard
```
Comment (abhijat) [inline build.sbt]: this case can be removed

File: src/main/scala/com.ds.practice/problema/EntryA.scala [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+import com.ds.practice.problema.consumer.{RequestConsumer, ScoreAggConsumer}
+import com.ds.practice.problema.http.HttpWrapper
+import com.ds.practice.problema.model.ScoreAggPayload
+import com.ds.practice.problema.util.{AppProperties, Constant}
+
+import scala.io.StdIn
+
+
+class EntryA() {
```
Comment (abhijat) [inline src/main/scala/com.ds.practice/problema/EntryA.scala]: probably dont need empty brackets here

File: src/main/scala/com.ds.practice/problema/EntryA.scala [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+  for (_ <- 1 to coresForAggregation) {
+    aggregatorPool.submit(new ScoreAggConsumer(associationScoreQueue))
+  }
+
+  /**
+   * Validates the incoming cmd and add the computed Request info to Request queue
+   * @param input CLI input
+   */
+  def validateAndSubmit(input: Array[String]): Unit = {
+    if (input.isEmpty || input.length < 2) {
```
Comment (abhijat) [inline src/main/scala/com.ds.practice/problema/EntryA.scala]: `imput.length < 2` covers `input.isEmpty` also?

File: src/main/scala/com.ds.practice/problema/consumer/ScoreAggConsumer.scala [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  /**
+   * Consumes the score and prints the output
+   * @param queueMsg - scores with Request meta info(needed to identify the request)
+   */
+  override def consume(queueMsg: ScoreAggPayload): Unit = {
+    val associationScore = queueMsg.scores
+    val avg = scoreAverage(associationScore)
+    val stdDeviation = scoreStandardDeviation(associationScore, avg)
+    val resultStr = s"${queueMsg.metaData} max=${associationScore.max} min=${associationScore.min} avg=$avg std-dev=$stdDeviation"
+    println(resultStr)
```
Comment (abhijat) [inline src/main/scala/com.ds.practice/problema/consumer/ScoreAggConsumer.scala:26]: can remove println if logging

Comment (deepakshingavi) [reply src/main/scala/com.ds.practice/problema/consumer/ScoreAggConsumer.scala:26]: no it is required

File: build.sbt [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+libraryDependencies ++= Seq(
+  "org.scalaj" % "scalaj-http_2.12" % "2.4.2",
+  "io.spray" %% "spray-json" % "1.3.5" % "provided",
+  "org.apache.spark" % "spark-core_2.12" % "2.4.4",
+  "org.apache.spark" % "spark-sql_2.12" % "2.4.4",
+  "org.scalatest" % "scalatest_2.12" % "3.3.0-SNAP2" % "provided",
+  "org.scalamock" % "scalamock_2.12" % "4.4.0"
+)
+
+assemblyMergeStrategy in assembly := {
```
Comment (abhijat) [inline build.sbt:16]: could be simply MergeStrategy.first rather than case statement

```
assemblyMergeStrategy in assembly := { MergeStrategy.first }
```

**Review by abhijat — APPROVED** [review]

**Pull request merged**